pkg/wireguard/agent/agent.go

```diff
@@ -109,6 +110,10 @@ func (a *Agent) Init() error {
 	link := &netlink.Wireguard{LinkAttrs: netlink.LinkAttrs{Name: types.IfaceName}}
 	err := netlink.LinkAdd(link)
 	if err != nil && !errors.Is(err, unix.EEXIST) {
+		if errors.Is(err, errOpNotSupported) {
+			return fmt.Errorf("wireguard not supported by the Linux kernel (netlink: %w). "+
+				"Please ensure your system has the wireguard kernel module installed", err)
```

Comment:
Would it be possible to provide the user on how they can achieve this? Is upgrading enough? If not, where can I find this module and installed? `apt-get update && apt-get install -y wireguard`? We can even have a short url that points to our docs with installation pointers if that's easier.

Reply:
So in mainline, you need kernel 5.6 and `CONFIG_WIREGUARD=m`. Some modern distros already ship Wireguard by default this way (Ubuntu 20.04 even has Wireguard backported to their older kernel). But other distros like AWS Bottlerocket have it disabled and therefore the user needs would have to recompile the kernel or try to install the module manually out-of-tree via DKMS.

So it's super hard to have a "one-size-fits-all" method, since it's very distro specific. We could link to https://www.wireguard.com/install/ maybe. That guide also installs the tools which we don't really need, but I guess they also don't hurt for troubleshooting. I'd like to avoid having to maintain our own installation instructions.

Reply:
Added a link to https://www.wireguard.com/install/